payload-functions.js

```diff
@@ -0,0 +1,153 @@
+const { helpers } = require("kaholo-aws-plugin");
+const _ = require("lodash");
+const { strToBase64 } = require("./helpers");
+
+function prepareCreateInstancePayload(params) {
+  if (params.maxCount < params.minCount) {
+    throw new Error("Max Count must be bigger or equal to Min Count");
+  }
+
+  const nameTag = [];
+  if (!_.isEmpty(params.NameTag)) {
+    nameTag.push({
+      Key: "Name",
+      Value: params.NameTag,
+    });
+  }
+
+  return {
+    ...(_.omit(params, "NameTag")),
+    UserData: strToBase64(params.UserData),
+    TagSpecifications: helpers.buildTagSpecification("instance", [params.TagSpecifications, ...nameTag]),
+  };
+}
+
+function prepareDescribeInstancesPayload(params) {
+  // EC2 does not allow MaxResults if InstanceIds are provided
+  if (params.InstanceIds.length) {
+    return _.omit(params, "MaxResults");
+  }
+
+  return params;
+}
+
+function prepareCreateVpcPayload(params) {
+  if (!params.CidrBlock && !params.AmazonProvidedIpv6CidrBlock) {
+    throw new Error("Must provide CIDR Block or select AmazonProvidedIpv6CidrBlock");
+  }
+
+  return {
+    ...(_.pick(params, ["CidrBlock", "AmazonProvidedIpv6CidrBlock", "InstanceTenancy", "DryRun"])),
+    TagSpecifications: helpers.buildTagSpecification("vpc", params.tags),
+  };
+}
+
+function prepareCreateInternetGatewayPayload(params) {
+  return {
+    DryRun: params.DryRun,
+    TagSpecifications: helpers.buildTagSpecification("internet-gateway", params.tags),
+  };
+}
+
+function prepareCreateRouteTablePayload(params) {
+  return {
+    VpcId: params.VpcId,
+    DryRun: params.DryRun,
+    TagSpecifications: helpers.buildTagSpecification("route-table", params.Tags),
+  };
+}
+
+function prepareCreateNatGatewayPayload(params) {
+  return {
+    ...(_.pick(params, ["SubnetId", "AllocationId", "DryRun"])),
+    TagSpecifications: helpers.buildTagSpecification("natgateway", params.Tags),
+  };
+}
+
+function validateAssociateRouteTableParams(params) {
+  if (!params.SubnetId && !params.GatewayId) {
+    throw new Error("You need to provide a Subnet ID or a Gateway ID!");
```

Comment:
`config.json`?

Reply:
Cannot be done in config.json -- either Subnet ID or Gateway ID need to be provided, none of those are strictly required, just one of them is.